resources/versioning.md

```diff
@@ -0,0 +1,89 @@
+# Versioning policy
+
+This page describes the versioning rules Meilisearch SDKs will follow once Meilisearch engine v1.0.0 is released and how/when we should increase the MAJOR, MINOR, and PATCH of the versions.
+
+## 🤖 Basic rules
+
+Meilisearch SDKs releases follow the [SemVer rules](https://semver.org/), including the following basic ones:
+
+> 🔥 Given a version number MAJOR.MINOR.PATCH, increment the:
+>
+> 1. MAJOR version when you make incompatible API changes
+> 2. MINOR version when you add functionality in a backwards compatible manner
+> 3. PATCH version when you make backwards compatible bug fixes
+
+
+**Changes that MAY lead the Meilisearch users (developers) to change their code are considered API incompatibility and will make us increase the MAJOR version of Meilisearch SDK.**
+
+**In other terms, if the users MAY have to do more steps than just downloading the new Meilisearch instance and running it, a new MAJOR is needed.**
+
+Examples of changes making the code break and then, involving increasing the MAJOR:
+
+- Any change related to the public API the SDK exposes like:
+    - Renaming a method name or classes
+    - Removing or changing a positional argument of a method.
+
+## Reasons why we should release a major version
+
+#### New runtimes adoption
+
+We don't have a strict policy yet regarding the bare minimum support of language version. Example:
+
+Ruby SDK has a `required_ruby_version >= 2.6` which prevents users to use the gem if they are running their apps with a `Ruby 2.5`.
+
+Is often good to only allow newer versions of the runtimes because they are maintained :sparkles: and there is always new features we can adopt in the code.
+
+But, there is a price on that, when we make that cut, we reduce the amount of users that will be able to adopt our SDK/service.
+
+List of current supported language version runtimes*:
+
+| Language/Framework | Version |
+| ------------------ | ------- |
+| .NET | netstandard2.0 |
+| Dart | >= 2.12.0 |
+| Go | >= 1.16 |
+| Java | >= 8.0 |
+| JavaScript | N/A |
+| Swift | >= 5.2 |
+| PHP | >= 7.4 |
+| Python | >= 3.7 |
+| Ruby | >= 2.6 |
+| Rust | N/A |
+
+When a SDK drops a minium version, we will release a new MAJOR version.
+
+\* This list may be outdated, always check the real version in the integration's repository.
+
+## Correlation between Meilisearch engine versioning and the SDKs
```

Comment:
That's good you approached this topic because this is one of my doubts.

I would like to have this "justification and communication" written somewhere, and also rules to be followed. Where do you think they should go (the justification and communication)?

Reply:
For me it's more about the form. You start by saying, "there is no plan to make SDK into v1"

This is not what the users want to know when they read your guide. They want to know 
- if the SDK versions will follow the version of Meilisearch or not 
- and then, which version of SDK is compatible with which version of Meilisearch, and how they can get the information

I would start to give your rules first, and then, the justifications, following by "we are open to feedback and your needs...blabla". Not the contrary